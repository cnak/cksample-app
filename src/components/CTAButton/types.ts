export interface CTAButtonProps {
    ctaText: string;
    onCTAPress: (value?: any) => void;
    style?: object;
    testID: string;
}
