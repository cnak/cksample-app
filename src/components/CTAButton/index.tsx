import React from 'react';
import styled from 'styled-components/native';

import StyledText from '../StyledText';
import { BUTTON_COLOUR, WHITE } from '../colors';

import { CTAButtonProps } from './types';

const Button = styled.TouchableOpacity`
    background-color: ${BUTTON_COLOUR};
    line-height: 1.25px;
    margin: 0 auto;
    padding: 8px 0;
    align-items: center;
    justify-content: center;
    border-radius: 40px;
    width: 50%;
`;

const CTAText = styled(StyledText)`
    color: ${WHITE};
`;

// TODO (nice-to-have?) yellow-border on long-press / on-focus?
// TODO accessibility traits
const CTAButton = ({ ctaText, onCTAPress, style, testID }: CTAButtonProps) => (
    <Button
        accessibilityRole="button"
        accessibilityLabel={ctaText}
        testID={testID}
        onPress={onCTAPress}
        activeOpacity={0.5}
        style={style}
    >
        <CTAText>{ctaText}</CTAText>
    </Button>
);

export default CTAButton;
