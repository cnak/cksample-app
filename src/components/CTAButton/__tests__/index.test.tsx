import React from 'react';
import renderer from 'react-test-renderer';

import CTAButton from '..';

describe('<CTAButton>', () => {
    it('matches snapshot', () => {
        const tree = renderer
            .create(<CTAButton ctaText="Tap here" onCTAPress={jest.fn()} testID="CTAButton:Title of Page" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
