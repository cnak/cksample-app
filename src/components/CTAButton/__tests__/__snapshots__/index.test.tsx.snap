// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CTAButton> matches snapshot 1`] = `
<View
  accessibilityLabel="Tap here"
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#00823B",
      "justifyContent": "center",
      "lineHeight": 1.25,
      "marginBottom": 60,
      "marginLeft": 0,
      "marginRight": 0,
      "marginTop": 20,
      "opacity": 1,
      "paddingBottom": 8,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 8,
      "shadowColor": "#003618",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 0,
    }
  }
  testID="CTAButton:Title of Page"
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Helvetica",
          "fontSize": 16,
        },
      ]
    }
  >
    Tap here
  </Text>
</View>
`;
